libs/src/services/discoveryProvider/index.js

```diff
@@ -28,14 +28,26 @@ class DiscoveryProvider {
 
   async init () {
     let endpoint
+    let pick
+
     if (this.autoselect) {
       endpoint = await this.autoSelectEndpoint()
     } else {
       if (!this.whitelist || this.whitelist.size === 0) {
         throw new Error('Must pass autoselect true or provide whitelist.')
       }
 
-      const pick = _.sample([...this.whitelist]) // selects random element from list.
+      const picks = await Utils.timeRequests([...this.whitelist].map(discprov => {
```

Comment:
This may not be what we want. This does Promise.all() not Promise.race()

Reply:
I don't know if this will work. This makes the entire startup take as long as the slowest discprov. (e.g. potentially forever).

Couple thoughts-
1. I think this code should live in the autoselect=true branch here (not false). It's nice to have a false option where we can fall back to a totally random selection & validation process.

2. The perhaps right way to make *this* code work is to do what we do in fetchCID. We dispatch a bunch of requests at once with a cancel token and cancel those that fail first. The first one to reply should be the discprov we pick.

3. The other tool we have that maybe solves this better (which fetchCID uses internally) is promiseFight, which is our own implementation of promise.any (subtly different than promise.race which will return a failing promise). promise.any hasn't made it into js yet.